### PR TITLE
CRAN release v1.4.1

### PR DESCRIPTION
**Pull request by krlmlr**
- [x] Resolve all tasks in `cran-comments.md`
- [x] Run `fledge::release()`
- [ ] When the package is accepted on CRAN, run `fledge::post_release()`

### CODE REVIEW

**Comment by krlmlr** [comment]
@szhorvat: I summarized the C core changes in https://github.com/igraph/rigraph/pull/680/files#diff-51920e95310ebfbc1ae31709f3b95f89afffbf4f1a6e38e8b2b406e2fb6197eaR19 . Please edit in this branch as necessary.

@ntamas: R-hub and winbuilder emails might arrive to you, I triggered the checks. Please let me know if there is anything suspicious.

Sending to CRAN when everything's good.

**Comment by ntamas** [comment]
@krlmlr Got a result from win-builder, all looks good there.

**Comment by szhorvat** [comment]
I edited the C core related part of the changelog to make it relevant for end users.

Looking at the current state of the changelog in general, I think it illustrates well a very common disconnect between "software developers" and the "end users" (i.e. the actual target audience). igraph is primarily intended for researchers. They should be able to judge at a glance what changes are relevant to _them_. This changelog show what is relevant to _you_, and includes multiple things that are entirely uninformative for just about any igraph user.

For example, "Chore" as a header name is meaningless to most people. It should not be used.

Every single item under "Chore" is completely irrelevant for any user. They should also be removed.

Going into this much detail under "Documentation" is again not useful. Just mention "documentation improvements", thay's enough.

"Don't ignore build/ when building the package because the vignette index is built there" is again irrelevant.

Generally, focus on _what_ is achieved, not _how_ it's achieved. Notice my edits for the C-core related changes. It does not matter that `isnan()` was replaced. What matters is that you can now use GCC 5 or earlier to build igraph. Similarly, _"Don't mention C++11 or C++17 for best compatibility with both newest R and older compilers, while still requesting a C++ compiler for linking."_ is not informative. Focus on the user visible changes: was it impossible to build 1.4.0 with older R versions (which)? If so, point out that now it's possible. As for the second part, with the linking, the relevant piece of information is: 1.4.0 could not be built on some systems due to failing to link to the C++ standard library. This was a regression in 1.4.0 and is now corrected.

To sum up,

 - talk to your target audience
 - make the changelog useful to them

**Comment by szhorvat** [comment]
I do understand the appeal of `fledge` to many software developers. Automating as much as possible is good, right? It's good for as long as it helps in getting a good end result. In this case, it encourages changelogs like what we see here, and actually makes it harder to write one what is useful and informative to our target audience. Let us please drop it.

**Comment by krlmlr** [comment]
Your suggestions are very valuable, would you mind if we borrowed them for fledge's own documentation?

I followed them in my most recent commit. We can drop the "Internal" section, but IMO it doesn't do much harm and still is helpful to more technical users and of course to us developers.

CC @maelle.

**Comment by maelle** [comment]
Thank you @szhorvat! These are interesting and legitimate concerns.

@krlmlr which part would you like to add to fledge docs, could you open an issue there? The remark on "Chore" reminds me of https://github.com/cynkra/fledge/issues/339

@szhorvat and others, let me mention a related idea supported by pkgdown just in case it might be relevant in the future:
- For an important release, one can write a [blog post _intended for users_](https://style.tidyverse.org/news.html#blog-post) and publish it "somewhere"*. 
- This release can then be added to the pkgdown configuration file so that the blog post is accessible from the "news" menu. See [pkgdown's own website](https://pkgdown.r-lib.org/) and [configuration](https://github.com/r-lib/pkgdown/blob/c354aa7e5ea1f9936692494c28c89e5bdd31fc68/pkgdown/_pkgdown.yml#L85). 

*:If that somewhere produces an RSS feed, possibly one that's specific by language, instead of release-watching this repository, an user could subscribe to that feed instead.

File: NEWS.md [hunk 1]
```diff
@@ -8,6 +8,14 @@
 
 - Fix errors when printing long vertex names (#677, @ahmohamed).
 
+- Fix regression that broke builds on some systems (e.g., GCC version 5 or earlier), introduced in igraph 1.4.0.
+
+## Documentation
+
+- Various improvements (#663, @maelle; #667).
```
Comment (maelle) [inline NEWS.md:15]: Isn't it this release that adds vignettes and a pkgdown website? If so, that'd be important to underline.

**Comment by szhorvat** [comment]
> Your suggestions are very valuable, would you mind if we borrowed them for fledge's own documentation?

Of course, feel free to use any of it.



> For an important release, one can write a [blog post _intended for users_](https://style.tidyverse.org/news.html#blog-post) and publish it "somewhere"*.

That would be great. The practical limitation is that someone has to take the time to write it ... As for where to publish, somewhere on igraph.org? The front page news section is set up like a blog, it's just not called that. The posts already have tags (C, Python, R, Mathematica), they are just not exposed on the website yet. Exposing them would make it easy for readers to isolate the R-specific posts.

**Comment by krlmlr** [comment]
I believe vignettes were added in igraph 1.4.0.

@ntamas: I just triggered the CRAN release, please either confirm (to release) or ignore that e-mail (to postpone, without further consequences).

File: NEWS.md (hunk 1 above)
Comment (krlmlr) [reply NEWS.md:15]: I believe vignettes were added in igraph 1.4.0.

**Comment by szhorvat** [comment]
@krlmlr Thanks for fixing up the changelog!

> We can drop the "Internal" section, but IMO it doesn't do much harm and still is helpful to more technical users and of course to us developers.

Yes, I agree. Nothing wrong with keeping it when it's arranged this way.

#### Some further suggestions

 - Point out in the **bugfixes** section that the linking error has been fixed (since this is something that users experienced directly, see #670 
 - Either remove mentions of Zenodo, or better: Create the Zenodo page now, before release, and include the master DOI into the main part of the changelog. @ntamas do you want to set up Zenodo, or do you want me to do it myself? Note: The actual Zenodo archive will only be created after a GitHub release is made, but I think the master DOI (not the specific DOI dor this one release) can be obtained earlier. If not, we can manually upload 1.4.0 to Zenodo to ensure that the master DOI is created.

    The motivation for all this is so that people can start citing R/igraph through its Zenodo entry, where all current team members are authors.  I expect that there are many more people who read the NEWS file than those who look at our GitHub page. Thus this is a good place to advertise the new citation option.

**Comment by krlmlr** [comment]
Added link to issues. Let's include the Zenodo link for this release if it's easy to set up. I'll rerun `fledge::release()` when done.

**Comment by maelle** [comment]
>  Exposing them would make it easy for readers to isolate the R-specific posts.

The feeds could even be tag specific too (I don't know how the website is built, though).

**Comment by krlmlr** [comment]
@ntamas: Tweaked, called `fledge::release()` again.

File: R/console.R [hunk 2]
```diff
@@ -20,10 +20,43 @@
 #
 ###################################################################
 
-.igraph.pb <- NULL
 
+#' The igraph console
+#'
+#' The igraph console is a GUI windows that shows what the currently running
```
Comment (szhorvat) [inline R/console.R:26]:
```suggestion
#' The igraph console is a GUI window that shows what the currently running
```

**Comment by ntamas** [comment]
CRAN pretests passed with the current release, I need to respond to Uwe that the "additional issues" have been fixed (have they? I think we need to tell them about the false positives with the newest GCC, right?).

Re Zenodo: I can do the setup tomorrow at the earliest. Since you already have a Zenodo account (as far as I know), it might be faster if you do it and just ping me when there's somethign that needs to be approved in the repository. If you haev no time for it today, that's okay, I'll do it tomorrow in the afternoon then.

**Comment by szhorvat** [comment]
> it might be faster if you do it and just ping me when there's somethign that needs to be approved in the repository

OK, will do.

**Comment by szhorvat** [comment]
@ntamas Found the discussion about the gengraph issue: You seem to be saying that there is no bug, just messy code. https://github.com/igraph/igraph/issues/2169

**Comment by krlmlr** [comment]
I suggest explaining that we fixed the LTO issues to our best knowledge.

I can postpone creation of the GitHub release until after you have finished the required setup.

**Comment by szhorvat** [comment]
> > it might be faster if you do it and just ping me when there's somethign that needs to be approved in the repository
> 
> OK, will do.

@ntamas I looked into this, and it can only be done by someone with "admin" access to the specific repo, which I don't have for `rigraph`. So it'll have to be you. I think all you really need to do is flip the relevant switch here: https://zenodo.org/account/settings/github/

**Comment by ntamas** [comment]
Hmmm, then I have to figure out how we did it the last time for the igraph repo :) I'll look into this, thanks for the heads up.
